### PR TITLE
feat: add cron go+java examples

### PR DESCRIPTION
**Pull request by safeer**

### CODE REVIEW

File: examples/go/echo/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module ftl/echo
 
-go 1.23.0
+go 1.23.1
```
Comment (stuartwdouglas) [inline examples/go/echo/go.mod:3]: Why this change?

Comment (stuartwdouglas) [reply examples/go/echo/go.mod:3]: It seems like almost all our tests and examples are on 1.23.0, if we are going to upgrade maybe we should do it all in one hit, although I don't know if there is any real reason to.

**Review by stuartwdouglas — APPROVED** [review]

File: examples/go/echo/go.mod (hunk 1 above)
Comment (safeer) [reply examples/go/echo/go.mod:3]: This was just to address a build error, but happy to leave as is:
```
info: Modules waiting to build: echo
error: initial deploy failed: failed to build module: failed to build module "echo": err: exit status 1: stderr: go: module ../../../.ftl/go/modules/cron listed in go.work file requires go >= 1.23.1, but go.work lists go 1.23.0; to update it:
	go work use
```
